src/components/List.js

```diff
@@ -1,12 +1,54 @@
 import React from 'react';
 import useFirestore from '../hooks/useFirestore';
+import { db } from '../lib/firebase';
 
 const List = ({ token }) => {
   const { docs } = useFirestore(token);
+
+  const checkPurchasedDate = (purchasedDate) => {
+    if (purchasedDate === null) {
+      return false;
+    }
+    const today = new Date();
+    let daySincePurchased = purchasedDate.toDate();
+
+    daySincePurchased.setDate(daySincePurchased.getDate() + 1); // Add one day to the last purchased date
+
+    // If today > daySincePurchased at least 24 hours have passed, return false to uncheck box
+    if (
+      Date.parse(today.toString()) >= Date.parse(daySincePurchased.toString())
+    ) {
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const handleCheckbox = async (event) => {
+    const currentDate = new Date();
```

Comment:
Non-rhetorical question for anyone and everyone - would it be preferable to save Date.now(), which would give us a unix timestamp, rather than `new Date()`, which appears to get converted into a Firebase timestamp (https://firebase.google.com/docs/reference/android/com/google/firebase/Timestamp)?
For me I feel like a unix timestamp would be more straightforward to deal with and leaves our logic slightly less tightly-coupled to firebase, but I don't know if there are advantages to doing it this away though that I'm not aware of.
If we were to use a Date.now(), in `checkPurchasedDate`, I believe that we would convert to a JS date with something that looks like `new Date(purchasedDate)`

Reply:
@MikeBLambert dates are making my head hurt also 😅 I'm going to step away for the night and think some more. Thanks for all the  comments!